lib/src/test/java/com/metaplex/lib/serialization/rpc/solana/SolanaRpcResponseTests.kt

```diff
@@ -0,0 +1,92 @@
+import com.metaplex.lib.experimental.serialization.serializers.rpc.solana.data
+import com.metaplex.lib.experimental.serialization.serializers.solana.AnchorAccountSerializer
+import com.metaplex.lib.modules.auctions.models.AuctionHouse
+import com.solana.core.PublicKey
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.modules.SerializersModule
+import kotlinx.serialization.modules.contextual
+import org.junit.Assert
+import org.junit.Test
+
+class SolanaRpcResponseTests {
+
+    val jsonResponseString = "{" +
+            "\"jsonrpc\":\"2.0\"," +
+            "\"result\":{" +
+            "\"context\":{" +
+            "\"apiVersion\":\"1.10.29\"," +
+            "\"slot\":149332850" +
+            "}," +
+            "\"value\":{" +
+            "\"data\":[" +
+            "\"KGzXa9VV9TC9WuTFjCbG0RR7CXM/o6Jb3sLJjqqmeNyI+uBYcn8mGLvtruiZlJOUc5O5XrLpZ+PbnWq5BpAKKCT7T9RhxMiSeA6Gza4qx8mdg9W+r6hT4lSELlTjkLzxG/JeSifEZqx4DobNrirHyZ2D1b6vqFPiVIQuVOOQvPEb8l5KJ8RmrAabiFf+q4GE+2h/Y0YYwDXaxDncGus7VZig8AAAAAABeA6Gza4qx8mdg9W+r6hT4lSELlTjkLzxG/JeSifEZqx4DobNrirHyZ2D1b6vqFPiVIQuVOOQvPEb8l5KJ8RmrP3+/MgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\"," +
+            "\"base64\"" +
+            "]," +
+            "\"executable\":false," +
+            "\"lamports\":4085520," +
+            "\"owner\":\"hausS13jsjafwWwGqZTUQRmWyvyxn9EQpqMwV1PBBmk\"," +
+            "\"rentEpoch\":345" +
+            "}" +
+            "}," +
+            "\"id\":1" +
+            "}"
+
+    val auctionHouse = AuctionHouse(
+        auctionHouseFeeAccount = PublicKey("DkAScnZa6GqjXkPYPAU4kediZmR2EESHXutFzR4U6TGs"),
+        auctionHouseTreasury = PublicKey("DebSyCbsnzMppVLt1umD4tUcJV6bSQW4z3nQVXQpWhCV"),
+        treasuryWithdrawalDestination = PublicKey("95emj1a33Ei7B6ciu7gbPm7zRMRpFGs86g5nK5NiSdEK"),
+        feeWithdrawalDestination = PublicKey("95emj1a33Ei7B6ciu7gbPm7zRMRpFGs86g5nK5NiSdEK"),
+        treasuryMint = PublicKey("So11111111111111111111111111111111111111112"),
+        authority = PublicKey("95emj1a33Ei7B6ciu7gbPm7zRMRpFGs86g5nK5NiSdEK"),
+        creator = PublicKey("95emj1a33Ei7B6ciu7gbPm7zRMRpFGs86g5nK5NiSdEK"),
+        bump = 253u,
+        treasuryBump = 254u,
+        feePayerBump = 252u,
+        sellerFeeBasisPoints = 200u,
+        requiresSignOff = false,
+        canChangeSalePrice = false,
+        escrowPaymentBump = 0u,
+        hasAuctioneer = false,
+        auctioneerPdaBump = 0u
+    )
+
+    val json = Json {
+        ignoreUnknownKeys = true
+        serializersModule = SerializersModule {
+            contextual(ContextualDataSerializer(AnchorAccountSerializer<AuctionHouse>()))
```

Comment:
This needs to be preconfigured? Can it be inferred from the type?

Reply:
Or can we configure multiple?

Reply:
in this case, yes it still needs to be pre configured. But this is something I am am trying to improve. Normally you don't need to configure like this, the system automatically figures out the correct serialize to use. This is a workaround needed for generic types, But I think i can improve it. 

And yes, multiple custom serializes can be configured.